lib/shanty_template_plugin/template_plugin.rb

```diff
@@ -0,0 +1,86 @@
+require 'deep_merge'
+require 'erubis'
+require 'fileutils'
+require 'shanty/plugin'
+require 'yaml'
+
+module ShantyTemplatePlugin
+  # Public: Shanty template plugin
+  class TemplatePlugin < Shanty::Plugin
+    PROJECT_EXTENSION = 'erb'
+    DS_EXTENSTION = 'syaml'
+
+    subscribe :build, :on_build
+    provides_config :datasource, 'plain'
+    provides_config :output_dir, 'build'
+    provides_projects_containing "**/*.#{PROJECT_EXTENSION}"
+
+    def on_build
+      load_data.each do |env, data|
```

Comment:
I think we should call this something other than `env` as this shadows the other `env`. Perhaps `template_env`?